### PR TITLE
Logging url rules

### PR DESCRIPTION
**Pull request by bashkarev**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | yes
| Breaks BC?    | yes
| Tests pass?   | yes
| Fixed issues  | https://github.com/yiisoft/yii2-debug/issues/173

### CODE REVIEW

File: framework/web/UrlManager.php [hunk 1]
```diff
@@ -263,7 +264,14 @@ public function parseRequest($request)
         if ($this->enablePrettyUrl) {
             /* @var $rule UrlRule */
             foreach ($this->rules as $rule) {
-                if (($result = $rule->parseRequest($this, $request)) !== false) {
+                $result = $rule->parseRequest($this, $request);
+                if (YII_DEBUG) {
+                    Yii::getLogger()->log([
+                        'rule' => ($rule instanceof UrlRule) ? (string)$rule : get_class($rule),
+                        'match' => $result !== false
+                    ], Logger::LEVEL_PROFILE, __METHOD__);
```
Comment (cebe) [inline framework/web/UrlManager.php]: why `LEVEL_PROFILE` and not `LEVEL_TRACE`?

Comment (bashkarev) [inline framework/web/UrlManager.php]: to exclude in Log panel

Comment (samdark) [inline framework/web/UrlManager.php]: Can't we exclude it by class/method name?

**Review by samdark — CHANGES_REQUESTED** [review]

File: framework/web/UrlManager.php (hunk 1 above)
Comment (bashkarev) [inline framework/web/UrlManager.php]: fixed

File: framework/CHANGELOG.md [hunk 2]
```diff
@@ -54,6 +54,7 @@ Yii Framework 2 Change Log
 - Enh #13074: Improved `\yii\log\SyslogTarget` with `$options` to be able to change the default `openlog` options. (timbeks)
 - Enh #13050: Added `yii\filters\HostControl` allowing protection against 'host header' attacks (klimov-paul)
 - Enh: Added constants for specifying `yii\validators\CompareValidator::$type` (cebe)
+- Enh: Added Logging url rules (bashkarev)
```
Comment (samdark) [inline framework/CHANGELOG.md]: Need `#13134` here

**Review by samdark — APPROVED** [review]
Except of missing issue number in changelog it looks fine to me.

File: framework/web/UrlRule.php [hunk 3]
```diff
@@ -126,6 +126,25 @@ class UrlRule extends Object implements UrlRuleInterface
      */
     private $_routeParams = [];
 
+    /**
+     * @return string
+     */
```
Comment (cebe) [inline framework/web/UrlRule.php]: `@since 2.0.11`

**Comment by cebe** [comment]
The good looks good so far but I wonder whether we might want to log the whole route object (or a serialization of it) to be able to display all relevant configurations in the debug panel. E.g. RestUrlRules or GroupUrlRule contain a lot more information that could be relevant in debug panel.

**Comment by cebe** [comment]
Form a progress point of view it would also be good to develop the debug panel in the framework changes together to make adjustments here when new aspects show up on toolbar side and merge them both after everything is finished.

**Review by SilverFire — APPROVED** [review]
Except minor comments by @cebe

**Comment by bashkarev** [comment]
??

**Comment by SilverFire** [comment]
> The good looks good so far but I wonder whether we might want to log the whole route object

The idea is good, but, probably, this amount of data could be too much for the debug file.

I think the PR should be accepted as is. Any objection @yiisoft/core-developers?

**Comment by samdark** [comment]
No objections from me.

**Comment by samdark** [comment]
Merged. Thanks!

**Pull request merged**